### PR TITLE
Add Xresources-based theme

### PR DESCRIPTION
**Pull request by SqrtMinusOne**
I use Xresources to configure themes. This PR adds themes called `xresources-light` and `xresources-dark`, provided that `xrdb` is found and `xrdb -query` returns the necessary colors.

This is a bit suboptimal for light theme (which I use), because light themes don't need as much shading as C-o and C-p currently do. But `solarized-light` also has this issue.

image

### CODE REVIEW

File: src/posting/app.py [hunk 1]
```diff
@@ -701,6 +702,52 @@ def on_mount(self) -> None:
         )
         self.theme_change_signal = Signal[ColorSystem](self, "theme-changed")
         self.theme = self.settings.theme
+        self.try_to_add_xresources_theme()
```
Comment (darrenburns) [inline src/posting/app.py:705]: I'd want to put this behind a config option `use_xresources` which is `False` by default. If that option is `True`, then this method would be called.

It should also be possible to set the theme to `xresources-dark` or `xresources-light` directly from the config file (e.g. `theme: xresource-dark`).

**Review by darrenburns — CHANGES_REQUESTED** [review]
I'd like this to sit behind a config option.

**Comment by darrenburns** [comment]
Thanks for the contribution! - I've done a short review, let me know what you think or if you need help.

**Comment by SqrtMinusOne** [comment]
Done. I've:
- add an option called `use_xresources`, as you suggested;
- changed the method to fail with more verbose errors in case something goes wrong;
- moved the invocation of the method from `on_mount` to the constructor. I'm not sure if there's a more suitable place for it, but it works.

**Comment by SqrtMinusOne** [comment]
Oops, my Emacs has auto-formatted the table in the README. I hope you don't mind.

**Comment by darrenburns** [comment]
Thanks! I'll try to review this this week.

**Comment by darrenburns** [comment]
Thank you - I've made a few minor changes. Moved stuff into a new module to keep it separate from the `App`, an simplified the code a bit.

Would you mind giving it a quick test? If it looks good to you I will merge.

**Comment by SqrtMinusOne** [comment]
Thanks.

The current version fails with:
```
  File "/home/pavel/00-Scratch/posting/src/posting/xresources.py", line 36, in load_xresources_themes
    missing_colors = set(XRDB_MAPPING.values()) - supplied_colors.keys()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'list'
```

The point of `list(itertools.chain(*self.XRDB_MAPPING.values()))` vas to flatten that list. I'll make a new PR with a fix.

**Pull request merged**